### PR TITLE
Feature/rtl fixed costs

### PR DESCRIPTION
**Pull request by peggles2**
This pull request adds React Testing Library for fixed costs.

### CODE REVIEW

File: solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js [hunk 1]
```diff
@@ -21,53 +22,53 @@ describe('FixedCosts', () => {
...
     render(<FixedCosts
       visible={true}
       goToStep={jest.fn()}
       setFixedCost={jest.fn()}
       totalFixedCosts='0'
       key='false' 
     />)
+  });
+
+  test('Has a heading caleld "Calculate your total fixed costs"', () => {
```
Comment (kscoulter) [inline solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js:35]: spelling mistake

File: solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js [hunk 2]
```diff
@@ -21,53 +22,53 @@ describe('FixedCosts', () => {
...
+    expect(screen.queryByRole('spinbutton', { name: /amortization/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /rent/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /insurance/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /salaries/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /utilities/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /deprecation/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /interest expense/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /property taxes/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /other monthly costs/i })).not.toBeInTheDocument()
+    expect(screen.queryByRole('spinbutton', { name: /other fixed costs/i })).not.toBeInTheDocument()
```
Comment (kscoulter) [inline solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js:59]: you can put a test id on the whole form and assert that it's not showing up, instead of testing for each input field to be hidden.

File: solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js [hunk 3]
```diff
@@ -81,109 +82,116 @@ describe('FixedCosts', () => {
     expect(screen.queryByRole('spinbutton', { name: /other fixed costs/i })).toBeInTheDocument()
   })
 
-  it('shows NumbersInputForm on no selection', () => {
-    const wrapper = shallow(<FixedCosts {...baseProps} />);
-    expect(wrapper.find('NumbersInputForm')).toHaveLength(0)
-    wrapper.setState({knowFixedCosts: 'no'})
-    expect(wrapper.find('NumbersInputForm')).toHaveLength(1)
-  })
+  test('shows the continue button when a selection has been made', () => {
```
Comment (kscoulter) [inline solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js:85]: good one

File: solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js [hunk 4]
```diff
@@ -81,109 +82,116 @@ describe('FixedCosts', () => {
...
-    const wrapper = shallow(
-      <FixedCosts {...baseProps} />
-    );
-    wrapper.setState({knowFixedCosts: 'no'})
-    wrapper.find('Form').simulate('submit')
+  test('shows NumbersInputForm on suggestion link click', () => {
+    const knowFixedCost = screen.getByRole('radio', { name: /yes, i know the total of my monthly fixed costs/i })
+    userEvent.click(knowFixedCost)
+    userEvent.click(screen.getByRole('button', { name: /add all fixed costs individually/i }))
+    expect(screen.queryByRole('spinbutton', { name: /total fixed cost/i })).not.toBeInTheDocument()
```
Comment (kscoulter) [inline solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js:111]: same comment here, put test id on the form and assert that the whole form is not present.

**Review by kscoulter — CHANGES_REQUESTED** [review]

File: solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js (hunk 2 above)
Comment (peggles2) [reply solution/ui/src/components/organisms/fixedCosts/fixedCosts.test.js:59]: This is a more detailed test. It also is testing none of these show up. The other way around is more generalized.

**Review by kscoulter — APPROVED** [review]
